lib/matplotlib/widgets.py

```diff
@@ -1417,6 +1417,543 @@ def _update(self):
+
+    'ctl+m' : Toggles the visibility of the ruler and text. 
+
+    Example
+    ----------
+
+    >>> xCoord = np.arange(0, 5, 1)
+    >>> yCoord = [0, 1, -3, 5, -3]
+    >>> fig = plt.figure()
+    >>> ax = fig.add_subplot(111)
+
+    >>> markerprops = dict(marker='o', markersize=5, markeredgecolor='red')
+    >>> lineprops = dict(color='red', linewidth=2)
+
+    >>> ax.grid(True)
+    >>> ax.plot(xCoord, yCoord)
+
+    >>> ruler = Ruler(ax=ax,
+                  useblit=True,
+                  markerprops=markerprops,
+                  lineprops=lineprops)
+
+    >>> plt.show()
+
+
+    """
+
+    def __init__(self,
+                 ax,
+                 active=True,
+                 length_unit=None,
+                 angle_unit='degree',
+                 print_text=False,
+                 useblit=False,
+                 lineprops=None,
+                 textprops=None,
+                 markerprops=None):
+        """
+        Add a ruler to *ax*. If ``ruler_active=True``, the ruler will be 
+        activated when the plot is first created. If ``ruler_unit`` is set the 
+        string will be appended to the length text annotations. 
+
+        """
+        AxesWidget.__init__(self, ax)
+
+        self.connect_events()
+
+        self.ax = ax
+        self.fig = ax.figure
+        self.canvas = ax.figure.canvas
+        self.print_text = print_text
+        self.visible = True
+        self.active = active
+        self.length_unit = length_unit
+        self.angle_unit = angle_unit
+        self.useblit = useblit and self.canvas.supports_blit
+
+        self.mouse1_pressed = False
+        self.mouse3_pressed = False
+        self.shift_pressed = False
+        self.control_pressed = False
+        self.end_a_lock = False
+        self.x0 = None
+        self.y0 = None
+        self.x1 = None
+        self.y1 = None
+        self.line_start_coords = None
+        self.line_end_coords = None
+        self.ruler_marker = None
+        self.background = None
+        self.ruler_moving = False
+        self.end_a_lock = False
+        self.end_b_lock = False
+        self.end_c_lock = False
+        self.old_marker_a_coords = None
+        self.old_marker_c_coords = None
+        self.old_mid_coords = None
+
+        if lineprops is None:
+            lineprops = {}
+
+        bbox = dict(facecolor='white',
+                    alpha=0.5,
+                    boxstyle='round',
+                    edgecolor='0.75')
+
+        default_textprops = dict(xy=(0, 1),
+                                 xytext=(10, -10),
+                                 xycoords='axes fraction',
+                                 textcoords='offset points',
+                                 ha='left',
+                                 va='center',
+                                 bbox=bbox)
+
+        x0 = self.ax.get_xlim()[0]
+        y0 = self.ax.get_ylim()[0]
+
+        self.ruler, = self.ax.plot([x0, x0], [y0, y0], label='ruler',
```

Comment:
I removed it. This is a leftover from when I attempted to use a legend to display the measure.